src/main/java/by/custom/utilcalculator/domain/tree/CommandTree.java

```diff
@@ -0,0 +1,83 @@
+package by.custom.utilcalculator.domain.tree;
+
+import by.custom.utilcalculator.domain.UserProgress;
+import by.custom.utilcalculator.domain.constants.steps.StepsIndicator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CommandTree {
+    private final Map<StepsIndicator, String> fieldsToCommands;
+    private final Node treeRoot;
+
+    private CommandTree() {
+        fieldsToCommands = HelperTree.fillMap();
+        treeRoot = HelperTree.buildTree();
+    }
+
+    public static CommandTree getInstance() {
+        return TreeHolder.TREE_HOLDER;
+    }
+
+    public Map<StepsIndicator, String> getFieldsToCommands() {
+        return fieldsToCommands;
+    }
+
+    public boolean validateCommand(final String requestingCommand, final UserProgress userProgress) {
+        return isRequestingCommandAcceptable(requestingCommand, getNode(userProgress));
+    }
+
+    public boolean isRequestingCommandAcceptable(String requestingCommand, Node node) {
```

Comment:
And do it every where. I'm pretty sure that it's possible to configure IDE to verify it
```suggestion
    public boolean isRequestingCommandAcceptable(final String requestingCommand, final Node node) {
```

Reply:
I haven't known about cleanup settings yet. Thanks a lot! This settings are awesome! 
![Code Cleanup Inspections_240415214758](https://github.com/Y-Pazniak/custom_calculator_chatbot/assets/73634869/cfb89694-3597-4984-8a7b-b7ba3a7f3e64)